unsloth/__init__.py

```diff
@@ -58,7 +58,13 @@
 # Try loading bitsandbytes and triton
 import bitsandbytes as bnb
 import triton
-from triton.common.build import libcuda_dirs
+triton_version = triton.__version__.split(".")
+triton_major = int(triton_version[0])
+if triton_major >= 3:
+    from triton.backends.nvidia.driver import libcuda_dirs
```

Comment:
This didn't work for me with `pytorch-triton==3.0.0+901819d2b6`:

```python
    import unsloth
  File "/home/carmocca/git/unsloth/unsloth/__init__.py", line 64, in <module>
    from triton.backends.nvidia.driver import libcuda_dirs
ImportError: cannot import name 'libcuda_dirs' from 'triton.backends.nvidia.driver' (/home/carmocca/git/nightly-venv/lib/python3.10/site-packages/triton/backends/nvidia/driver.py)
```

Also, you would want to update L98 which also imports this once there' a working solution

Reply:
Soz on the delay - I made it a try except